src/core/lib/iomgr/error.c

```diff
@@ -286,6 +296,11 @@ grpc_error *grpc_error_set_str(grpc_error *src, grpc_error_strs which,
   return new;
 }
 
+const char *grpc_error_get_str(grpc_error *err, grpc_error_strs which) {
+  if (is_special(err)) return NULL;
```

Comment:
`grpc_error_set_str` doesn't treat special errors differently, but this getter does.